doc/optimizers.md

```diff
@@ -152,6 +152,170 @@ optimizer.Optimize(f, coordinates);
  * [Adam: A Method for Stochastic Optimization](http://arxiv.org/abs/1412.6980)
  * [Differentiable separable functions](#differentiable-separable-functions)
 
+## AdamR
+
+*An optimizer for [differentiable separable functions](#differentiable-separable-functions).*
+
+AdamR is a modified version of Adam with warm restarts. This allows the optimiser to have a automated decreasing
+step size without modification. AdamR uses Cyclical Decay as its default decay policy.
+
+#### Constructors
+
+ * `AdamR()`
+ * `AdamR(`_`stepSizeMax, stepSizeMin, epochRestart, multFactor, batchSize`_`)`
+ * `AdamR(`_`stepSizeMax, stepSizeMin, epochRestart, multFactor, batchSize, beta1, beta2, eps, maxIterations, tolerance, shuffle`_`)`
+ * `AdamR(`_`stepSizeMax, stepSizeMin, epochRestart, multFactor, batchSize, beta1, beta2, eps, maxIterations, tolerance, shuffle, resetPolicy`_`)`
+
+Note that the `AdamR` class is based on the `AdamRType<`_`UpdateRule`_`>` class
+with _`UpdateRule`_` = AdamUpdate`.
+
+#### Attributes
+
+| **type** | **name** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| `double` | **`stepSizeMax`** | Intitial step size for each iteration. | `0.02` |
+| `double` | **`stepSizeMin`** | Final step size for each iteration. | `0.01` |
+| `size_t` | **`epochRestart`** | Size of initial iteration. | `50` |
+| `double` | **`multFactor`** | Rate at which iteration size changes. | `2.0` |
+| `size_t` | **`batchSize`** | Number of points to process in a single step. | `32` |
+| `double` | **`beta1`** | Exponential decay rate for the first moment estimates. | `0.9` |
+| `double` | **`beta2`** | Exponential decay rate for the weighted infinity norm estimates. | `0.999` |
+| `double` | **`eps`** | Value used to initialize the mean squared gradient parameter. | `1e-8` |
+| `size_t` | **`max_iterations`** | Maximum number of iterations allowed (0 means no limit). | `100000` |
+| `double` | **`tolerance`** | Maximum absolute tolerance to terminate algorithm. | `1e-5` |
+| `bool` | **`shuffle`** | If true, the function order is shuffled; otherwise, each function is visited in linear order. | `true` |
+| `bool` | **`resetPolicy`** | If true, parameters are reset before every Optimize call; otherwise, their values are retained. | `true` |
+
+The attributes of the optimizer may also be modified via the member methods
+ `StepSize()`, `BatchSize()`, `Beta1()`, `Beta2()`, `Eps()`, `MaxIterations()`, `UpdatePolicy()`, `DecayPolicy()`
+ `Tolerance()`, `Shuffle()`, and `ResetPolicy()`.
```

Comment:
I know this is pedantic but I have always been careful to make sure the ordering of options here matches the ordering in the table above.  Can you make the same change here please?